front/app/components/HookForm/SelectionChoiceSetup/index.tsx

```diff
@@ -0,0 +1,242 @@
+import { DndProvider } from 'react-dnd-cjs';
+import HTML5Backend from 'react-dnd-html5-backend-cjs';
+import { get } from 'lodash-es';
+
+// react hook form
+import { Controller, useFieldArray, useFormContext } from 'react-hook-form';
+
+// components
+import {
+  Box,
+  Label,
+  Button,
+  LocaleSwitcher,
+  Icon,
+  Input,
+} from '@citizenlab/cl2-component-library';
+import { SectionField } from 'components/admin/Section';
+import { List, SortableRow } from 'components/admin/ResourceList';
+import Error, { TFieldName } from 'components/UI/Error';
+
+// Typings
+import { Locale, CLError, RHFErrors } from 'typings';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { generateTempId } from 'components/FormBuilder/components/FormBuilderSettings/utils';
+
+interface Props {
+  name: string;
+  onSelectedLocaleChange?: (locale: Locale) => void;
+  locales: Locale[];
+  allowDeletingAllOptions?: boolean;
+  platformLocale: Locale;
+  fieldLabel: string | JSX.Element | null;
+  addButtonLabel: string | JSX.Element | undefined;
+}
+
+const SelectionChoiceSetup = ({
+  onSelectedLocaleChange,
+  name,
+  locales,
+  allowDeletingAllOptions = false,
+  platformLocale,
+  fieldLabel,
+  addButtonLabel,
+}: Props) => {
+  const {
+    control,
+    formState: { errors: formContextErrors },
+    setValue,
+    trigger,
+  } = useFormContext();
+  const [selectedLocale, setSelectedLocale] = useState<Locale | null>(
+    platformLocale
+  );
+
+  // Handles locale change
+  useEffect(() => {
+    setSelectedLocale(platformLocale);
+    onSelectedLocaleChange?.(platformLocale);
+  }, [platformLocale, onSelectedLocaleChange]);
+  const handleOnSelectedLocaleChange = useCallback(
+    (newSelectedLocale: Locale) => {
+      setSelectedLocale(newSelectedLocale);
+      onSelectedLocaleChange?.(newSelectedLocale);
+    },
+    [onSelectedLocaleChange]
+  );
+
+  // Handles drag and drop
+  const { move } = useFieldArray({
+    name,
+  });
+  const handleDragRow = (fromIndex: number, toIndex: number) => {
+    move(fromIndex, toIndex);
+  };
+
+  // Handles add and remove options
+  const addOption = (value, name: string) => {
+    const newValues = value;
+    newValues.push({
+      title_multiloc: {},
+    });
+    setValue(name, newValues);
+  };
+  const removeOption = (value, name, index) => {
+    const newValues = value;
+    newValues.splice(index, 1);
+    setValue(name, newValues);
+  };
+
+  const defaultOptionValues = [];
+  const errors = get(formContextErrors, name) as RHFErrors;
+  const apiError = errors?.error && ([errors] as CLError[]);
+  const validationError = errors?.message;
+
+  if (isNilOrError(platformLocale) || isNilOrError(locales)) {
+    return null;
+  }
+
+  if (selectedLocale) {
+    return (
+      <>
+        <Controller
+          name={name}
+          control={control}
+          defaultValue={defaultOptionValues}
+          render={({ field: { ref: _ref, value: choices, onBlur } }) => {
```

Comment:
Would be nice to type `choices` here

Reply:
It looks like none of the "value" attributes for hook form components are typed ( I don't think you can?). I did add a type for when it's being mapped/used though! :+1: